ECS.h

```diff
@@ -361,6 +363,16 @@ namespace ECS
 			ComponentHandle<T> component;
 		};
 
+		// Called when a component is removed
+		template<typename T>
+		struct OnComponentRemoved
```

Comment:
This needs to be fired for each component in `Entity::removeAll()` in addition to `Entity::remove<T>()`.

Reply:
I just realized that doing so will require a more complex change since you don't have access to the template variable `<T>` in `removeAll()`. One solution I might suggest is adding a new pure virtual function to `Internal::BaseComponentContainer` called `void componentRemoved(Entity* entity)`, and in `ComponentContainer<T>` you can implement the method like so:

```cpp
virtual void componentRemoved(Entity* ent)
{
    auto handle = ComponentHandle<T>(this);
    ent->getWorld()->emit<Events::OnComponentRemoved>({ent, this});
}
```

You would then call `componentRemoved()` in both `removeAll()` and `remove<T>()`.